src/guides/v2.4/extension-dev-guide/security/content-security-policies2.4.md

```diff
@@ -0,0 +1,282 @@
+---
+group: php-developer-guide
+title: Content Security Policies
+---
+
+Content Security Policies (CSP) are a powerful tool to mitigate against Cross Site Scripting (XSS)
+and related attacks, including card skimmers, session hijacking, clickjacking, and more. Web servers
+send CSPs in response HTTP headers (namely `Content-Security-Policy` and
+`Content-Security-Policy-Report-Only`) to browsers that whitelist the origins of scripts, styles,
+and other resources. Together, CSPs and built-in browser features help prevent:
+
+*  Loading a malicious script from an attacker's website
+*  A malicious inline script from sending credit card info to an attacker's website
+*  Loading a malicious style that will make users click on an element that wasn't supposed to be on a page
+
+See [Content Security Policy (CSP)](https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP)
+and [Content-Security-Policy](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy)
+to learn more about CSP and each individual policy.
+
+## Magento and CSP
+
+As of version 2.3.5, Magento supports CSP headers and provides ways to configure them. (This
+functionality is defined in the Magento_Csp module.) Magento also provides default configurations at
+the application level and for individual core modules that require extra configuration. Policies can
+be configured for `adminhtml` and `storefront` areas separately to accommodate different use cases.
+Magento also permits configuring unique CSPs for specific pages.
+
+CSP can work in two modes:
+
+*  `report-only` - In this mode, Magento reports policy violations but does not interfere. This mode isuseful for debugging.  By default, CSP violations are written to the browser console, but they can be configured to be reported to an endpoint as an HTTP request to collect logs. There are a number of services that will collect, store, and sort your store's CSP violations reports for you.
+
+*  `restrict mode` - In this mode, Magento acts on any policy violations.
+
+## Default configuration
+
+By default, CSP is configured in `report-only` mode, which allows merchants and developers to
+configure policies to work according to their custom code. After the policies have been configured,
+switch the mode to `restrict`.
+
+Once configured, Magento can enforce policies like these:
+
+*  Any resource, such as `.js`, `.css`, `.jpg`, or `.ttf` files, can only be loaded from the store's domain
+*  Iframes can only include pages from the store itself
+*  AJAX requests can only be sent to the store
+*  Forms can only be sent to the store
+*  Only whitelisted inline scripts and styles to be compiled by browser
+
+For more details check the `Magento/Csp/etc/config.xml` file. Some domains have already been
+whitelistedfor modules that require it. For instance if the `Magento_Paypal` module is installed,
```

Comment:
```suggestion
whitelisted for modules that require it. For examplle, if the `Magento_Paypal` module is installed,
```

Reply:
double L in example

Reply:
oops. i'll fix it when i change the file name. you can disregard this suggestion.